### PR TITLE
Migrate Server vault UpstreamAuthority to V1

### PR DESCRIPTION
**Pull request by MarcosDY**

### CODE REVIEW

File: pkg/server/plugin/upstreamauthority/vault/vault_client.go [hunk 1]
```diff
@@ -250,18 +251,18 @@ func (c *ClientConfig) configureTLS(vc *vapi.Config) error {
...
 		c, err := tls.LoadX509KeyPair(c.clientParams.ClientCertPath, c.clientParams.ClientKeyPath)
 		if err != nil {
-			return fmt.Errorf("failed to parse client cert and private-key: %w", err)
+			return status.Errorf(codes.InvalidArgument, "failed to parse client cert and private-key: %v", err)
 		}
 		clientCert = c
 		foundClientCert = true
 	case c.clientParams.ClientCertPath != "" || c.clientParams.ClientKeyPath != "":
-		return fmt.Errorf("both client cert and client key are required")
+		return status.Errorf(codes.InvalidArgument, "both client cert and client key are required")
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/vault/vault_client.go:259]:
```suggestion
		return status.Error(codes.InvalidArgument, "both client cert and client key are required")
```

File: pkg/server/plugin/upstreamauthority/vault/vault_client.go [hunk 2]
```diff
@@ -293,39 +294,39 @@ func (c *Client) Auth(path string, body map[string]interface{}) (*vapi.Secret, e
...
 
 	id, err := secret.TokenID()
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "unable to get TokenID: %v", err)
 	}
 	renewable, err := secret.TokenIsRenewable()
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "unable to get token is renewable: %v", err)
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/vault/vault_client.go:325]:
```suggestion
		return nil, status.Errorf(codes.Internal, "unable to determine if token is renewable: %v", err)
```

File: pkg/server/plugin/upstreamauthority/vault/vault_client.go [hunk 3]
```diff
@@ -354,28 +355,28 @@ func (c *Client) SignIntermediate(ttl string, csr *x509.CertificateRequest) (*Si
 	path := fmt.Sprintf("/%s/root/sign-intermediate", c.clientParams.PKIMountPoint)
 	s, err := c.vaultClient.Logical().Write(path, reqData)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "fails to sign intermediate: %v", err)
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/vault/vault_client.go:358]:
```suggestion
		return nil, status.Errorf(codes.Internal, "failed to sign intermediate: %v", err)
```

File: pkg/server/plugin/upstreamauthority/vault/vault_test.go [hunk 4]
```diff
@@ -5,105 +5,99 @@ import (
...
+		wantAuth                 AuthMethod
+		wantNamespaceIsNotNil    bool
+		envKeyVal                map[string]string
+		expectToken              string
+		expectCertAuthMountPoint string
+		expectClientCertPath     string
+		expectClientKeyPath      string
+		AppRoleAuthMountPoint    string
+		AppRoleID                string
+		AppRoleSecretID          string
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/vault/vault_test.go:57]:
```suggestion
		appRoleAuthMountPoint    string
		appRoleID                string
		appRoleSecretID          string
```

File: pkg/server/plugin/upstreamauthority/vault/vault_test.go [hunk 5]
```diff
@@ -112,426 +106,586 @@ func (vps *VaultPluginSuite) Test_Configure() {
...
+	}
 }
 
-func (vps *VaultPluginSuite) Test_MintX509CA_InvalidVaultResponse() {
-	vps.fakeVaultServer.LookupSelfResponse = []byte(testLookupSelfResponse)
-	vps.fakeVaultServer.LookupSelfResponseCode = 200
-	vps.fakeVaultServer.SignIntermediateReqEndpoint = "/v1/test-pki/root/sign-intermediate"
-	vps.fakeVaultServer.SignIntermediateResponseCode = 200
-	vps.fakeVaultServer.SignIntermediateResponse = []byte(testInvalidSignIntermediateResponse)
+func TestMintX509CA_InvalidCSR(t *testing.T) {
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/vault/vault_test.go:571]: Why is this not part of the TestMintX509CAFails test above?

File: pkg/server/plugin/upstreamauthority/vault/vault.go [hunk 6]
```diff
@@ -131,52 +143,46 @@ func (p *Plugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi
 	p.authMethod = am
 	p.cc = vcConfig
 
-	return &spi.ConfigureResponse{}, nil
+	return &configv1.ConfigureResponse{}, nil
 }
 
-func (p *Plugin) MintX509CA(req *upstreamauthorityv0.MintX509CARequest, stream upstreamauthorityv0.UpstreamAuthority_MintX509CAServer) error {
+func (p *Plugin) MintX509CAAndSubscribe(request *upstreamauthorityv1.MintX509CARequest, stream upstreamauthorityv1.UpstreamAuthority_MintX509CAAndSubscribeServer) error {
```
[the remark this answers is not in the document]
Comment (MarcosDY) [reply pkg/server/plugin/upstreamauthority/vault/vault.go:149]: I'll update to req

File: pkg/server/plugin/upstreamauthority/vault/vault_test.go (hunk 5 above)
Comment (MarcosDY) [reply pkg/server/plugin/upstreamauthority/vault/vault_test.go:636]: refactored now to be into the same unit test

**Review by azdagron — APPROVED** [review]
\o/

**Pull request merged**